zookeeper-server/src/main/java/org/apache/zookeeper/cli/SyncCommand.java

```diff
@@ -55,18 +61,27 @@ public CliCommand parse(String[] cmdArgs) throws CliParseException {
     @Override
     public boolean exec() throws CliException {
         String path = args[1];
+        CompletableFuture<Integer> cf = new CompletableFuture<>();
+
         try {
             zk.sync(path, new AsyncCallback.VoidCallback() {
-
                 public void processResult(int rc, String path, Object ctx) {
-                    out.println("Sync returned " + rc);
+                    cf.complete(rc);
                 }
             }, null);
+
+            try {
+                int resultCode = cf.get(CONNECTION_TIMEOUT, TimeUnit.MILLISECONDS);
+                out.println("Sync returned " + resultCode);
+            } catch (TimeoutException ex) {
+                out.println("Sync is timeout within " +  CONNECTION_TIMEOUT + " ms");
+            }
         } catch (IllegalArgumentException ex) {
             throw new MalformedPathException(ex.getMessage());
+        } catch (InterruptedException | ExecutionException ex) {
```

Comment:
About catching `InterruptedException`: http://www.javapractices.com/topic/TopicAction.do?Id=251
We don't do anywhere else in our code which maybe not a good practice.
What do you think @BELUGABEHR ?

Reply:
I thought about it.
In this specific case we don't care. The CLI will exit.

Reply:
I follow an example from this:
https://github.com/apache/zookeeper/blob/master/zookeeper-server/src/main/java/org/apache/zookeeper/cli/SetQuotaCommand.java#L82

Reply:
Got it. Makes sense.

Reply:
When you catch the Interrupted exception, it clears the Thread's state.  I would separate the catch into two blocks and be sure to reinstate the flag to 'interrupted' so that any other code will also quickly exit.

```
} catch (InterruptedException ie) {
    Thread.currentThread().interrupt();
   throw new CliWrapperException(ie);
}
```

Reply:
Yes but in this case it is not really needed.
We can add it but not so important

Reply:
I think it's always best to do it the "right" way... it will insulate the code from changes later when it may become needed.

Reply:
It's missing from everywhere else in the code unfortunately.
There's no harm adding it here anyway.

Reply:
Yep, better to add.
Better to file a JIRA for fixing it every where